// @progress/kendo-react-layout/dist/npm/card/Avatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var Enums_1 = require("./interfaces/Enums");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var package_metadata_1 = require("../package-metadata");
var Avatar = function (props) {
    kendo_licensing_1.validatePackage(package_metadata_1.packageMetadata);
    return (React.createElement("div", { style: props.style, className: kendo_react_common_1.classNames('k-avatar', "k-avatar-" + props.fill, "k-avatar-" + props.shape, "k-avatar-" + props.themeColor, {
            'k-avatar-bordered': props.border,
            'k-avatar-sm': props.size === 'small',
            'k-avatar-md': props.size === 'medium',
            'k-avatar-lg': props.size === 'large'
        }, props.className) },
        React.createElement("span", { className: "k-avatar-" + props.type }, props.children)));
};
Avatar.propTypes = {
    className: PropTypes.string,
    shape: PropTypes.oneOf(['circle', 'square', 'rounded', 'rectangle']),
    type: PropTypes.oneOf(['text', 'image', 'icon']),
    fill: PropTypes.oneOf(['solid', 'outline']),
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    themeColor: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'info', 'success', 'warning', 'error', 'dark', 'light', 'inverse', 'inherit'])
};
Avatar.defaultProps = {
    type: Enums_1.avatarType.TEXT,
    shape: Enums_1.avatarShape.SQUARE,
    size: Enums_1.avatarSize.MEDIUM,
    fill: Enums_1.avatarFill.SOLID,
    themeColor: Enums_1.avatarThemeColor.PRIMARY
};
exports.default = Avatar;
